core/src/main/scala/kafka/log/ProducerIdMapping.scala

```diff
@@ -189,22 +193,36 @@ class ProducerIdMapping(val topicPartition: TopicPartition,
     }
   }
 
+  def checkForExpiredPids(currentTimeMs: Long) {
+    pidMap.retain { case (pid, lastEntry) =>
+      currentTimeMs - lastEntry.timestamp < maxPidExpirationMs
```

Comment:
Since `maxPidExpirationMs` is global and static, do we want to maintain it in each mapping, or could we just refer to it in `Log` only where load / checkForExpire can access?